membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -70,4 +82,35 @@ class AttributeController extends Controller with LazyLogging {
 
   def membership = lookup("membership", identity[Attributes])
   def features = lookup("features", onSuccess = Features.fromAttributes, onNotFound = Some(Features.unauthenticated))
+  
+  def updateAttributes(identityId : String): Action[AnyContent] = backendForSyncWithZuora.async { implicit request =>
+
+    val tp = request.touchpoint
+
+    val result: EitherT[Future, String, Attributes] =
+      for {
+        contact <- EitherT(tp.contactRepo.get(identityId).map(_ \/> s"No contact for $identityId"))
+        memSubF = EitherT[Future, String, Option[Subscription[SubscriptionPlan.Member]]](tp.subService.current[SubscriptionPlan.Member](contact).map(a => \/.right(a.headOption)))
+        conSubF = EitherT[Future, String, Option[Subscription[SubscriptionPlan.Contributor]]](tp.subService.current[SubscriptionPlan.Contributor](contact).map(a => \/.right(a.headOption)))
+        memSub <- memSubF
+        conSub <- conSubF
+        _ <- EitherT(Future.successful(if (memSub.isEmpty && conSub.isEmpty) \/.left("No paying relationship") else \/.right(())))
+        attributes = Attributes( UserId = identityId, Tier = memSub.map(_.plan.charges.benefit.id), MembershipNumber = contact.regNumber, Contributor = conSub.map(_.plan.charges.benefit.id))
+        res <- EitherT(tp.attrService.update(attributes).map(\/.right))
+      } yield attributes
+
+    result.run.map(_.fold(
+      error => {
+        logger.error(s"Failed to update attributes - $error")
+        ApiErrors.badRequest(error)
+      },
+      attributes => {
+        logger.info(s"${attributes.UserId} -> ${attributes.Tier} || ${attributes.Contributor}")
```

Comment:
it's not coming in for a year! also info only goes to the instance's disk, cloudwatch, and membership-logs kibana, which I believe are all access controlled and encrypted.